lib/pengine/unpack.c

```diff
@@ -2790,6 +2797,11 @@ unpack_migrate_to_failure(pe_resource_t *rsc, pe_node_t *node, xmlNode *xml_op,
          */
         target_lrm_resource != NULL
         && first_named_child(target_lrm_resource, XML_LRM_TAG_RSC_OP) != NULL
+        /* If there's any probe/monitor operation on the target newer than this
+         * failed migrate_to indicating it was not running there, this migrate_to
+         * failure no longer matters for the target.
+         */
+        && ((target_monitor == NULL) || (is_newer_op(target_monitor, xml_op) <= 0))
```

Comment:
I'm also a little concerned that we're introducing a tighter dependency on synchronized clocks, but again it seems reasonable and I don't have a better idea.

Reply:
Indeed. Functionalized as well now.